build_tools/cmake/ctest_all.sh

```diff
@@ -62,6 +62,10 @@ if [[ "${IREE_VULKAN_F16_DISABLE}" == 1 ]]; then
   label_exclude_args+=("^vulkan_uses_vk_khr_shader_float16_int8$")
 fi
 
+if [[ "$(uname)" == "Darwin" ]]; then
+  label_exclude_args+=("^exclude-os=macos$")
+fi
```

Comment:
There is already a mechanism for excluding tests on Windows just below this (labels weren't a good option since a few tests within test suites are/were failing). As much as I like disabling tests where they are defined, I'd prefer we not add too many ways to do the same thing.

Reply:
Hmm, I kinda like the using labels better, as you said it stays with the test definition. Splitting it out in a secret place makes it hard to discover. But right it is less fine grained. So fine to me to switch to the existing way.

Reply:
Yeah, more tradeoffs... we should really burn down the list of failing tests (by fixing them), particularly considering how many are failing on Windows. At least the list of failures is small on macOS.

Reply:
Totally. I was trying to debug the remaining failing test but it's a bit involved to me. Given they aren't core runtime or compiler failures, I just disabled them in order to put up the CI flow first. Will continue looking into fixing them properly.

Reply:
Yeah I agree both on trying to do this consistently and usually preferring the failure be identified with the test. OTOH having a single list of all the things failing is nice too